components/SyncReport/SyncReport.tsx

```diff
@@ -46,6 +46,10 @@ const SyncReport: React.FunctionComponent<SyncReportProps> = ({ closeModal }) =>
   }, [syncingStatusReport.lastBlockServer]);
 
   useEffect(() => {
+    // for some reason when you run the sync process the first block you got (end_block) (yes `end_block`, this is not a mistake)
+    // is one more that the wallet height, if it's the first time you got the wallet birthday + 1
+    // because of that I need, for the math to add 1 in the birthday.
+    // but in the screen I show the real wallet birthday.
```

Comment:
could a transaction be missed in the very first block? if a wallet has a transaction on its birthday when just created?

Reply:
As you know, I'm not the best for answering this, but... in my experience, when I create a new wallet always have to sync a number of blocks, I don't remember, maybe 99 blocks or something... said that, It seems improbable that a new wallet have a tx in the first block. But I'm sure that @zancas or @AloeareV have more valid opinions about this.

Reply:
If your question is:  "Why does a brand new wallet sync?"
Then this is the answer:

For a brand new wallet the client:
1.  Asks lightwalletd what it believe the [latest tip is](https://github.com/zingolabs/zingo-mobile/blob/da04bb79fcec031a831b3496f33214e34e92cf13/rust/lib/src/lib.rs#L24).
2.  [Subtracts 100 from that](https://github.com/zingolabs/zingo-mobile/blob/da04bb79fcec031a831b3496f33214e34e92cf13/rust/lib/src/lib.rs#L33).
3. Uses [that number](https://github.com/zingolabs/zingo-mobile/blob/da04bb79fcec031a831b3496f33214e34e92cf13/rust/lib/src/lib.rs#L40) as a sync start point for the wallet so that it has some buffer in the case of a "reorg" (i.e.  a blockchain split).

I feel uncertain about this architecture, and think it might be vulnerable to a malicious `lightwalletd`.   Am I not _sure_ this is relevant to your question.

Reply:
ok so if we use a <birthday + 1> then there will still be a 99 block buffer?

Reply:
I think we are safe then... if I understand well.

Reply:
No, I am sorry I added confusion.   The case that adds 100 blocks is only for a new wallet.

After looking at the `get_birthday` code, I think we need to expose an interface to `zingo-mobile` that gets the actual birthday (if that's what's needed).

Reply:
in zingo-mobile I get the birthday from this call: `RPCModule.execute('seed', '');` which have two fields: `seed` & `birthday`...

Reply:
But to be clear... I run this: `await RPCModule.execute('sync', '');` no parameters at all... and when I run this: `RPCModule.execute('syncstatus', '');` I noticed that with the first batch info the `end_block` is equal to `birthday + 1`. This info from `syncstatus` doesn't mean nothing wrong with the `sync` process execute before....
This conversation is really interesting but IMO don't have nothing to do with this PR...